icu4j/main/classes/core/src/com/ibm/icu/text/DateTimePatternGenerator.java

```diff
@@ -2649,6 +2649,25 @@ DateTimeMatcher set(String pattern, FormatParser fp, boolean allowDuplicateField
                 if (subField > 0) subField += value.length();
                 type[field] = subField;
             }
+
+            // #20739, we have a skeleton with milliseconde, but no seconds
+            if (!original.isFieldEmpty(FRACTIONAL_SECOND) && original.isFieldEmpty(SECOND)) {
+                // Force the use of seconds
+                for (int i = 0; i < types.length; ++i) {
+                    int[] row = types[i];
+                    if (row[1] == SECOND) {
+                        // first entry for SECOND
+                        original.populate(SECOND, (char)row[0], (char)row[3]);
```

Comment:
I wouldn't nitpick about this if I weren't already leaving another comment, but you only need to cast `row[0]`, not `row[3]`.

Reply:
I'll fix it if you think I should fix the first one.